### DISCUSSION TITLE
Setup hashring for thanos receiver that is running on Kubernetes using helm chart

### DISCUSSION THREAD

**Discussion by hsdp-aas** [thanos-io/thanos#5082] [Questions & Answers]
Hi Team,

We have configured thanos on Kubernetes using Helm chart https://github.com/bitnami/charts/tree/master/bitnami/thanos . The setup has been installed successfully, but when sending data from prometheus instances to thanos we have observed that the data capture in thanos query is not consistent. We have observed gaps in graph also. Upon investigating we got to know that we need to setup hashring instead of using loadbalancer to capture the data at the thanos receiver end. We have 3 thanos receiver replicas.

We also know that we can use https://github.com/observatorium/thanos-receive-controller to enable dynamics hashring, but would need help in integrating the same in thanos helm chart.

Is there any parameters that we can enable to in the helm chart to enable dynamic hashring?

Thanks in advance.

**Comment by yeya24** [comment]
> Is there any parameters that we can enable to in the helm chart to enable dynamic hashring?

I don't think there is any chart to do this. You need to add the receive controller to the helm chart.

If you would like to try jsonnet then you can take a look at https://github.com/observatorium/observatorium repo. It also provides some example yaml files so you can try to add the manifests to the chart.

**Reply by hsdp-aas** [reply]
Thanks for the reply @yeya24. https://github.com/observatorium/thanos-receive-controller. We did try this but couldn't update the statefull sets in helm. If there's any documentation to update the receiver controller into helm would be helpful.

**Reply by yeya24** [reply]
I don't think there is any doc for this. To add it to your chart, you just need to add the controller yaml to your chart and then I think that's it.
